dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/storage/SeedStorage.scala

```diff
@@ -0,0 +1,243 @@
+          FIRST_WALLET_TIME
+        case Failure(exception) => throw exception
+      }
+      Try {
+        val ivString = json(IV).str
+        val cipherTextString = json(CIPHER_TEXT).str
+        val rawSaltString = json(SALT).str
+        (ivString, cipherTextString, rawSaltString, creationTimeNum)
+      } match {
+        case Success(value)     => Right(value)
+        case Failure(exception) => throw exception
+      }
+    }
+
+    val encryptedEither: Either[ReadMnemonicError, EncryptedMnemonic] =
+      readJsonTupleEither.flatMap {
+        case (rawIv, rawCipherText, rawSalt, rawCreationTime) =>
+          val encryptedOpt = for {
+            iv <- ByteVector.fromHex(rawIv).map(AesIV.fromValidBytes)
+            cipherText <- ByteVector.fromHex(rawCipherText)
+            salt <- ByteVector.fromHex(rawSalt).map(AesSalt(_))
+          } yield {
+            logger.debug(
+              s"Parsed contents of $seedPath into an EncryptedMnemonic")
+            EncryptedMnemonic(AesEncryptedData(cipherText, iv),
+                              salt,
+                              Instant.ofEpochSecond(rawCreationTime))
+          }
+          val toRight: Option[Right[ReadMnemonicError, EncryptedMnemonic]] =
+            encryptedOpt
+              .map(Right(_))
+
+          toRight.getOrElse(
+            Left(JsonParsingError("JSON contents was not hex strings")))
+      }
+    encryptedEither
+  }
+
+  /**
+    * Reads the wallet mnemonic from disk and tries to parse and
+    * decrypt it
+    */
+  def decryptMnemonicFromDisk(
+      seedPath: Path,
+      passphrase: AesPassword): DecryptedMnemonic = {
+
+    val encryptedEither = readEncryptedMnemonicFromDisk(seedPath)
+
+    val decryptedEither: Either[ReadMnemonicError, DecryptedMnemonic] =
+      encryptedEither.flatMap { encrypted =>
+        encrypted.toMnemonic(passphrase) match {
+          case Failure(exc) =>
+            logger.error(s"Error when decrypting $encrypted: $exc")
+            Left(ReadMnemonicError.DecryptionError)
+          case Success(mnemonic) =>
+            logger.debug(s"Decrypted $encrypted successfully")
+            val decryptedMnemonic =
+              DecryptedMnemonic(mnemonic, encrypted.creationTime)
+            Right(decryptedMnemonic)
+        }
+      }
+
+    decryptedEither match {
+      case Left(err)    => sys.error(err.toString)
+      case Right(value) => value
+    }
+  }
+
+  def getPrivateKeyFromDisk(
+      seedPath: Path,
+      passphrase: AesPassword,
+      bip39PasswordOpt: Option[String]): ExtPrivateKey = {
+    val mnemonic = decryptMnemonicFromDisk(seedPath, passphrase).mnemonicCode
+    val seed = bip39PasswordOpt match {
+      case Some(pw) =>
+        BIP39Seed.fromMnemonic(mnemonic = mnemonic, password = pw)
+      case None =>
+        BIP39Seed.fromMnemonic(mnemonic = mnemonic,
+                               password = BIP39Seed.EMPTY_PASSWORD)
+    }
+
+    seed.toExtPrivateKey(SegWitMainNetPriv)
```

Comment:
ExtKeyPrivVersion should be a parameter to `getPrivateKeyFromDisk()` or readable from the appConfig